wallet/src/repository/wallet/mod.rs

```diff
@@ -328,16 +329,7 @@ where
 
         // Total amount of state and db transactions
         let total = state.transaction_next_id + u32::try_from(state.local_movements.len()).unwrap();
-        let mut transactions: Vec<model::BalanceMovement> = Vec::with_capacity(total as usize);
-
-        // Prepend local pending balance movements (not yet included in blocks)
-        let local_movements = state.local_movements.values().cloned().collect::<Vec<_>>();
-        transactions.extend_from_slice(&local_movements);
-
-        // Prepend balance movements of pending blocks
-        state.pending_movements.values().for_each(|x| {
-            transactions.extend_from_slice(x);
-        });
+        let mut keyed_transactions: BTreeMap<u32, model::BalanceMovement> = BTreeMap::new();
```

Comment:
This is a `BTreeMap` because it needs to be sorted, why does it need to be sorted?

Reply:
It is a nice to have, as in that ways clients will retrieve all transactions having some order.